### PR TITLE
Android 12 support

### LINKED ISSUE #7: Android 12 support
App with your library crashes on Android 12 with "`Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.`"

### PR DESCRIPTION
**Pull request by zigapovhe**
Added support for Android12. Fixes issue https://github.com/romellfudi/FudiNFC/issues/7

### CODE REVIEW

**Comment by romellfudi** [comment]
Hi @zigapovhe , do you test in previous version for instance 9 , 10, 11?

**Comment by zigapovhe** [comment]
> Hi @zigapovhe , do you test in previous version for instance 9 , 10, 11?

I didn't even properly test it for Android 12, that's why I've sent you an email. Not sure if you understand me correctly. 🙂

**Review by romellfudi — CHANGES_REQUESTED** [review]
I can't build the project, Does it necessary to update kotlin & compileSdkVersion versions?

File: app/build.gradle [hunk 1]
```diff
@@ -5,15 +5,15 @@ apply plugin: 'kotlin-kapt'
 
 android {
 
-    compileSdkVersion 29
-    buildToolsVersion '29.0.2'
+    compileSdkVersion 31
```
Comment (romellfudi) [inline app/build.gradle:8]: I can't build the project, Does it necessary to update kotlin & compileSdkVersion versions?

File: build.gradle [hunk 2]
```diff
@@ -1,12 +1,12 @@
 
 buildscript {
-    ext.kotlin_version = '1.5.0'
+    ext.kotlin_version = '1.6.10'
```
Comment (romellfudi) [inline build.gradle:3]: I can't build the project, Does it necessary to update kotlin & compileSdkVersion versions?

File: fudi-nfc/build.gradle [hunk 3]
```diff
@@ -26,12 +26,12 @@ ext {
     allLicenses = ["Apache-2.0"]
 }
 android {
-    compileSdkVersion 29
-    buildToolsVersion "29.0.2"
+    compileSdkVersion 31
```
Comment (romellfudi) [inline fudi-nfc/build.gradle:29]: I can't build the project, Does it necessary to update kotlin & compileSdkVersion versions?

File: fudi-nfc/src/main/java/com/romellfudi/fudinfc/gear/NfcAct.kt [hunk 4]
```diff
@@ -92,7 +92,7 @@ abstract class NfcAct : AppCompatActivity(), CreateNdefMessageCallback {
             this,
             0,
             Intent(this, javaClass).addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP),
-            0
+            PendingIntent.FLAG_MUTABLE
```
Comment (romellfudi) [inline fudi-nfc/src/main/java/com/romellfudi/fudinfc/gear/NfcAct.kt:95]: Why didn't you add a clause condition for higher Build.VERSION_CODES.M?

**Comment by romellfudi** [comment]
@zigapovhe  please indicate your distribution Version:

`distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-all.zip`

**Comment by romellfudi** [comment]
@zigapovhe  I dont have a Android 12 and the changes does not work on Android 10. Thus I will crate a release branch with your PR

**Review by romellfudi — APPROVED** [review]
On the grounds the PR is only for Android 12 devices, thus it works correctly

**Comment by romellfudi** [comment]
Thanks very much @zigapovhe

**Pull request merged**